File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        
        for (int c = 0; c < commands.length; c++) {
            ArrayList<Integer> list = new ArrayList<>();
            
            for (int i = commands[c][0] - 1; i < commands[c][1]; i++) {
                list.add(array[i]);
            }
            
            Collections.sort(list);
            answer[c] = list.get(commands[c][2] - 1);
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.kt
class Solution {
    fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
        var answer = intArrayOf()
        for((i, j, k) in commands) {
            answer += array.sliceArray(i-1 until j).sorted()[k-1]
        }
        return answer
    }
}